File: Categories/Tree/530_minimumAbsoluteDifferenceInBST.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    private int minDiff = Int32.MaxValue;
    private int prev = -1;

    private void InorderTraversal(TreeNode root) {
        if (root.left != null) {
            InorderTraversal(root.left);
        }

        if (prev != -1) {
            minDiff = Math.Min(
                minDiff,
                Math.Abs(prev - root.val)
            );
        }
        prev = root.val;

        if (root.right != null) {
            InorderTraversal(root.right);
        }
    }

    public int GetMinimumDifference(TreeNode root) {
        InorderTraversal(root);

        return minDiff;       
    }
}

File: Daily/2023/530_minimumAbsoluteDifferenceInBST.py
from typing import *


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder_traversal(self, root):
        if root.left is not None:
            self.inorder_traversal(root.left)
        
        if self.prev != -1:
            self.min_diff = min(
                self.min_diff,
                abs(root.val - self.prev)
            )
        self.prev = root.val

        if root.right is not None:
            self.inorder_traversal(root.right)
            

    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        INF = float('inf')
        self.min_diff = INF
        self.prev = -1

        self.inorder_traversal(root)

        return self.min_diff
        